src/gretel_trainer/trainer.py

```diff
@@ -0,0 +1,156 @@
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel("DEBUG")
+
+
+class ExtendedEnum(Enum):
+    """Utility class for Model enum"""
+
+    @classmethod
+    def get_types(cls):
+        return list(map(lambda c: c.name, cls))
+
+    @classmethod
+    def get_config(cls, model: str):
+        return cls[model].config
+
+
+class Model(namedtuple("Model", "config"), ExtendedEnum):
+    """Enum to pair valid models and configurations"""
+
+    GretelLSTM = "templates/gretel_lstm.yaml"
+    GretelCTGAN = "templates/gretel_ctgan.yaml"
+
+
+class Trainer:
+    """Automated generative training and sampling tool
+
+    Args:
+        project_name (str, optional): Gretel project name. Defaults to "trainer".
+        max_header_clusters (int, optional): Max number of clusters per batch. Defaults to 20.
+        max_rows (int, optional): Max number of rows per batch. Defaults to 50000.
+        model_type (str, optional): Options include ["GretelLSTM", "GretelCTGAN"]. Defaults to "GretelLSTM".
+        model_kwargs (dict, optional): Modify model configuration settings by key. E.g. {'epochs': 20}
+    """
+
+    def __init__(
+        self,
+        project_name: str = "trainer",
+        max_header_clusters: int = 20,
+        max_rows: int = 50000,
+        model_type: str = "GretelLSTM",
+        model_kwargs: dict = {}
+    ):
+
+        configure_session(api_key="prompt", cache="yes", validate=True)
+
+        self.df = None
+        self.dataset_path = None
+        self.project_name = project_name
+        self.project = create_or_get_unique_project(name=project_name)
+        self.max_header_clusters = max_header_clusters
+        self.max_rows = max_rows
+
+        if model_type in Model.get_types():
+            config = pkgutil.get_data(__name__, Model.get_config(model_type))
+            self.config = yaml.load(config, Loader=yaml.FullLoader)
+
+            # Update default config settings with kwargs by key
+            for key, value in model_kwargs.items():
+                self.config = self.replace_nested_key(self.config, key, value)
+            logger.debug(json.dumps(self.config,indent=2))
+
+        else:
+            raise ValueError(
+                f"Invalid model type. Must be {Model.get_model_types()}")
+
+    def train(self, dataset_path: str, overwrite: bool = True, round_decimals: int = 4):
+        """Train a model on the dataset
+
+        Args:
+            dataset_path (str): Path or URL to CSV
+            overwrite (bool, optional): Overwrite previous progress. Defaults to True.
+            round_decimals (int, optional): Round decimals in CSV as preprocessing step. Defaults to 4.
+        """
+        self.dataset_path = dataset_path
+        self.df = self._preprocess_data(
+            dataset_path=dataset_path, round_decimals=round_decimals
+        )
+        self.run = self._initialize_run(df=self.df, overwrite=overwrite)
+        self.run.train_all_partitions()
+
+    def load(self):
```

Comment:
Nit: this could probably be a classmethod that actually serves as a constructor. It wasn't obvious to me how to use this and it should probably take the project name as a param since you are using that as part of the manifest name.